texar/torch/utils/variables.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2018 The Texar Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Utility functions related to variables.
+"""
+
+# pylint: disable=invalid-name
+
+__all__ = [
+    "add_variable",
+    "collect_trainable_variables"
+]
+
+
+def add_variable(variable, var_list):
+    """Adds variable to a given list.
+
+    Args:
+        variable: A (list of) variable(s).
+        var_list (list): The list where the :attr:`variable` are added to.
+    """
+    if isinstance(variable, (list, tuple)):
+        for var in variable:
+            add_variable(var, var_list)
+    else:
+        # Checking uniqueness gives error
+        # if variable in var_list:
```

Comment:
Why is this line commented?

Reply:
Removed. Since it is a list, and pytorch variables do not have a name associated to check for uniqueness, checking if a variable exists in a list throws an error.